### ISSUE TITLE
Why substitute handler returns null?

### ISSUE THREAD

**Issue by miyconst** [Starcounter/Starcounter#2821]
I have a substitute handler:

```cs
Json json = Self.GET<Json>("/some-url-out-of-current-app", () => {
    return new ContainerPage();
});

sectionJson.Rows.Add(json);
```

Where `ContainerPage` is a Json View Model class, `sectionJson` is an instance of a Json View Model class, `Rows` is an `Arr<Json>`.

In this example `json` variable is always `NULL`, regardless of what substitute action returns.

cc @chrhol, @warpech

**Comment by alemoi** [comment]
Thanks for the question. I should probably look at this. Do you have your project online or easy steps to reproduce?

**Comment by miyconst** [comment]
I pushed full source code to a branch: https://github.com/Polyjuice/Website/tree/playground

The solution is a bit of a mess right now, but should be enough to see the error. Set a break point here: https://github.com/Polyjuice/Website/blob/playground/Website/Website/Program.cs#L91 and navigate to `http://localhost:8080/website/team`.

**Comment by alemoi** [comment]
Big thanks for reporting. The issue should be solved via 8444e23bfbab0c752b32567f9312e745c3c576ae
The problem was in static file handler changes: static file handler now responds with 404 non-null response which was considered a normal response. Now I have to filter out 404 internal response specifically, which seems to work fine now. Let me know!

**Comment by miyconst** [comment]
Confirm the fix in the 76 & 78 builds, the 77 has a bug.
